include/Graph/Graph.hpp

```diff
@@ -25,6 +25,7 @@
 #include <limits.h>
 
 #include <atomic>
+#include <cassert>
```

Comment:
why including `cassert`?

Reply:
In general we have managed these cases with a negative return value, without assert.
Could you use the same mechanism?